tracer/src/Datadog.Trace/AspNet/TracingHttpModule.cs

```diff
@@ -75,13 +75,24 @@ public void Dispose()
         {
         }
 
-        internal static void AddHeaderTagsFromHttpResponse(System.Web.HttpContext httpContext, Scope scope)
+        internal static void AddHeaderTagsFromHttpResponse(HttpContext httpContext, Scope scope)
         {
-            if (httpContext != null && HttpRuntime.UsingIntegratedPipeline && _canReadHttpResponseHeaders && !Tracer.Instance.Settings.HeaderTags.IsNullOrEmpty())
+            if (!Tracer.Instance.Settings.HeaderTags.IsNullOrEmpty())
             {
-                try
+                WrapResponseHeadersAction(httpContext, () =>
```

Comment:
It looks like `WrapResponseHeadersAction()` was added to wrap any arbitrary code that accesses response headers and catch any exceptions. It's a neat way of not writing the `try/catch/log` code twice, but I'm not sure it's worth the additional overhead here. IIRC, this lambda will allocate a new `Action` on each call, and there's also the indirection of executing an `Action` instead of a direct method call.

Then again, both calls to `WrapResponseHeadersAction()` sit behind settings that are disabled by default ("copy headers to tags" and "enabled ASM") so maybe it's ok?